src/main/java/com/cloudogu/scm/review/MergeObstacleCheckHook.java

```diff
@@ -0,0 +1,150 @@
+
+  private final DefaultPullRequestService pullRequestService;
+  private final MergeService mergeService;
+  private final MessageSenderFactory messageSenderFactory;
+
+  @Inject
+  public MergeObstacleCheckHook(DefaultPullRequestService pullRequestService, MergeService mergeService, MessageSenderFactory messageSenderFactory) {
+    this.pullRequestService = pullRequestService;
+    this.mergeService = mergeService;
+    this.messageSenderFactory = messageSenderFactory;
+  }
+
+  @Subscribe(async = false)
+  public void checkForObstacles(PreReceiveRepositoryHookEvent event) {
+    if (!pullRequestService.supportsPullRequests(event.getRepository())) {
+      return;
+    }
+    List<PullRequest> pullRequests = pullRequestService.getAll(event.getRepository().getNamespace(), event.getRepository().getName());
+    new Worker(event).process(pullRequests);
+  }
+
+  private class Worker {
+    private final Repository repository;
+    private final HookBranchProvider branchProvider;
+    private final HookMergeDetectionProvider mergeDetectionProvider;
+    private final MessageSender messageSender;
+
+    private Worker(PreReceiveRepositoryHookEvent event) {
+      this.repository = event.getRepository();
+      this.branchProvider = event.getContext().getBranchProvider();
+      this.mergeDetectionProvider = event.getContext().getMergeDetectionProvider();
+      this.messageSender = messageSenderFactory.create(event);
+    }
+
+    private void process(List<PullRequest> pullRequests) {
+      pullRequests.forEach(this::process);
+    }
+
+    private void process(PullRequest pullRequest) {
+      if (hasStatusOpen(pullRequest)) {
+        processOpen(pullRequest);
+      } else {
+        LOG.debug("ignoring pull request {}, because it in status {}", pullRequest.getId(), pullRequest.getStatus());
+      }
+    }
+
+    private void processOpen(PullRequest pullRequest) {
+      if (branchesAreModified(pullRequest) && isMerged(pullRequest)) {
+        try {
+          Collection<MergeObstacle> mergeObstacles = mergeService.verifyNoObstacles(PermissionCheck.mayPerformEmergencyMerge(repository), repository, pullRequest);
+          if (!mergeObstacles.isEmpty()) {
+            LOG.info("about to merge pull request #{} for repository {} with {} violated rule(s) by push", pullRequest.getId(), repository.getNamespaceAndName(), mergeObstacles.size());
+            String[] message =
+              {
+                format("This merges pull request #%s (%s -> %s) with %s violated rule(s).", pullRequest.getId(), pullRequest.getSource(), pullRequest.getTarget(), mergeObstacles.size()),
+                "This merge is permitted only because you have the permission for emergency merges",
+                "or the violated rules are not blocking ones.",
+                "Please check this pull request:"
+              };
+            messageSender.sendMessageForPullRequest(pullRequest, message);
```

Comment:
It looks like the pull request is not marked as emergency merged. The merge status does not have the exclamation mark in the ui.